diff --git a/dist/chunk-DJNTPCIJ.js b/dist/chunk-DJNTPCIJ.js
index 0e8a0e73381252d06332f8ab3609f3878e633afb..90eb91882516442ea4dab08f9e4a7dc6419f3e58 100644
--- a/dist/chunk-DJNTPCIJ.js
+++ b/dist/chunk-DJNTPCIJ.js
@@ -325,7 +325,8 @@ var queryGenerator = createReactGenerator({
       file: getFile(operation, { pluginKey: [pluginZodName] }),
       schemas: getSchemas(operation, { pluginKey: [pluginZodName], type: "function" })
     };
-    if (!isQuery || isMutation) {
+    const forceInclude = options.query?.forceInclude?.(operation);
+    if (!forceInclude && (!isQuery || isMutation)) {
       return null;
     }
     return /* @__PURE__ */ jsxs(
@@ -434,7 +435,8 @@ var mutationGenerator = createReactGenerator({
     } = useApp();
     const oas = useOas();
     const { getSchemas, getName, getFile } = useOperationManager();
-    const isQuery = !!options.query && options.query?.methods.some((method) => operation.method === method);
+    const forceIncludeQuery = options.query?.forceInclude?.(operation);
+    const isQuery = forceIncludeQuery || !!options.query && options.query?.methods.some((method) => operation.method === method);
     const isMutation = !isQuery && difference(options.mutation ? options.mutation.methods : [], options.query ? options.query.methods : []).some((method) => operation.method === method);
     const importPath = options.mutation ? options.mutation.importPath : "@tanstack/react-query";
     const mutation = {
@@ -465,7 +467,8 @@ var mutationGenerator = createReactGenerator({
       name: getName(operation, { type: "const", suffix: "MutationKey" }),
       typeName: getName(operation, { type: "type", suffix: "MutationKey" })
     };
-    if (!isMutation) {
+    const forceInclude = options.mutation?.forceInclude?.(operation);
+    if (!forceInclude && !isMutation) {
       return null;
     }
     return /* @__PURE__ */ jsxs(
diff --git a/dist/types-CQseu_km.d.cts b/dist/types-CQseu_km.d.cts
index e26bd56f916ea61189914c6fac195320d0207468..a858a13dfe4ea1524037cff9482099014a3f6916 100644
--- a/dist/types-CQseu_km.d.cts
+++ b/dist/types-CQseu_km.d.cts
@@ -124,6 +124,7 @@ type QueryKey = Transformer;
  */
 type MutationKey = Transformer;
 type Query = {
+    forceInclude?: (operation: Operation) => boolean;
     /**
      * Define which HttpMethods can be used for queries
      * @default ['get']
@@ -139,6 +140,7 @@ type Query = {
     importPath?: string;
 };
 type Mutation = {
+    forceInclude?: (operation: Operation) => boolean;
     /**
      * Define which HttpMethods can be used for mutations
      * @default ['post', 'put', 'delete']
diff --git a/dist/types-CQseu_km.d.ts b/dist/types-CQseu_km.d.ts
index e26bd56f916ea61189914c6fac195320d0207468..a858a13dfe4ea1524037cff9482099014a3f6916 100644
--- a/dist/types-CQseu_km.d.ts
+++ b/dist/types-CQseu_km.d.ts
@@ -124,6 +124,7 @@ type QueryKey = Transformer;
  */
 type MutationKey = Transformer;
 type Query = {
+    forceInclude?: (operation: Operation) => boolean;
     /**
      * Define which HttpMethods can be used for queries
      * @default ['get']
@@ -139,6 +140,7 @@ type Query = {
     importPath?: string;
 };
 type Mutation = {
+    forceInclude?: (operation: Operation) => boolean;
     /**
      * Define which HttpMethods can be used for mutations
      * @default ['post', 'put', 'delete']
